anta/tests/routing/generic.py

```diff
@@ -268,3 +268,80 @@ def test(self) -> None:
             # Verifying that the specified IPv4 routes are of the expected type.
             if expected_route_type != (actual_route_type := route_data.get("routeType")):
                 self.result.is_failure(f"{entry} - Incorrect route type - Expected: {expected_route_type} Actual: {actual_route_type}")
+
+
+class VerifyRouteEntry(AntaTest):
+    """Verifies the route entries of the IPv4 prefixes.
+
+    This test performs the following checks for each IPv4 prefix:
+
+      1. Verifies the specified BGP route exists in the routing table.
+      2. For each specified next-hop:
+          - Verifies a path with matching next-hop exists.
+          - Supports `strict: True` to verify that only the specified nexthops by which routes are learned, requiring an exact match.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if routes exist with paths matching the expected next-hops.
+    * Failure: The test will fail if:
+        - A route entry is not found for given ipv4 prefixes.
+        - A path with specified next-hop is not found.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.routing:
+      generic:
+        - VerifyRouteEntry:
+            route_entries:
+                - prefix: 10.10.0.1/32
+                  vrf: default
+                  nexthops:
+                    - 10.100.0.8
+                    - 10.100.0.10
+    ```
+    """
+
+    categories: ClassVar[list[str]] = ["routing"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show ip route vrf all", revision=4)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyRouteEntry test."""
+
+        route_entries: list[IPv4Routes]
+        """List of IPv4 route(s)."""
+
+        @field_validator("route_entries")
+        @classmethod
+        def validate_route_entries(cls, route_entries: list[IPv4Routes]) -> list[IPv4Routes]:
+            """Validate that 'nexthops' field is provided in each route entry."""
+            for entry in route_entries:
+                if entry.nexthops is None:
+                    msg = f"{entry} 'nexthops' field missing in the input"
+                    raise ValueError(msg)
+            return route_entries
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyRouteEntry."""
+        self.result.is_success()
+
+        output = self.instance_commands[0].json_output
+
+        for entry in self.inputs.route_entries:
+            # Verify if the prefix exists in route table
+            if (route_data := get_value(output, f"vrfs..{entry.vrf}..routes..{entry.prefix}", separator="..")) is None:
+                self.result.is_failure(f"{entry} - prefix not found")
+                continue
+
+            # Verify the nexthop addresses.
```

Comment:
With a period or without a period, pick one for all comments :P Usually comments do no have a period.
```suggestion
            # Verify the nexthop addresses
```

Reply:
Updated